### PR TITLE
Do not set WOOCOMMERCE_BLOCKS_PHASE inside build-zip script.

### PR DESCRIPTION
**Pull request by samueljseay**
### Changes proposed in this Pull Request:

The `bin/build-zip.sh` script is not always run in a prod setting so we should not set the `WOOCOMMERCE_BLOCKS_PHASE` env variable inside the script unless it has not been set already. Now in the place we need to override it (live branch workflow) we pass in the override value so experimental blocks can be built.

In future this variable should be removed, but for now keeping mention of it in the script is an ok interim step.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. You can check if experimental blocks are available in the live branch for this PR. Create a JN site from the PR, go to block editor and try add an experimental block like "Collection Filter". You can also check the contents of the blocks.ini file that is bundled with the plugin to see the value is correct.
2. Testing the other builds will require that you fork this repo, setup actions and run this. Or if you feel more comfortable with the simplicity of these changes, you could simulate a local test of the change by running `WOOCOMMERCE_BLOCKS_PHASE=1 ./bin/build-zip.sh` and see that the resulting zip has the correct blocks.ini value.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message

Do not set WOOCOMMERCE_BLOCKS_PHASE in the build-zip.sh script, pass it when calling the script in relevant workflows.

#### Comment 

</details>

### CODE REVIEW

File: .github/workflows/build-release-zip-file.yml [hunk 1]
```diff
@@ -1,38 +1,40 @@
...
-      ref:
-        description: 'By default the zip file is generated from the branch the workflow runs from, but you can specify an explicit reference to use instead here (e.g. refs/tags/tag_name or refs/heads/release/x.x). The resulting file will be available as an artifact on the workflow run.'
-        required: false
-        default: ''
+    workflow_dispatch:
+        inputs:
+            ref:
+                description: 'By default the zip file is generated from the branch the workflow runs from, but you can specify an explicit reference to use instead here (e.g. refs/tags/tag_name or refs/heads/release/x.x). The resulting file will be available as an artifact on the workflow run.'
+                required: false
+                default: ''
```
Comment (samueljseay) [inline .github/workflows/build-release-zip-file.yml:8]: Prettier autoformatted these files. Sorry for the noise

**Review by dinhtungdu — COMMENTED** [review]
@samueljseay Tested and can confirm the PR does its job. But I wonder how should we handle the case when running the build zip script locally? I think having a default env var set to `1` will be nice and more flexible.

**Comment by samueljseay** [comment]
@dinhtungdu I don't think we can make any assumption on that. What would be your general purpose of running build zip locally you think? If there is a use case one may simply include the variable when running the script. We could document that in a README I suppose? I'm not going to rush to introduce some pnpm script shortcut because those clutter the codebase and often get out of date quickly when they're not used by anyone.

> I think having a default env var set to 1 will be nice and more flexible.

i'm not sure what you mean. this is the opposite of flexible for me just because internally setting variables that control the build make the script very unflexible to be called from command line. Its as if an argument to your function is overridden internally every time you call it.

I guess the issue is that in core we have a proper feature gating system that isn't dependent specifically on one variable. I think in future we'd just get rid of blocks.ini and use feature gating to ensure that experimental blocks are not in the release?

**Comment by dinhtungdu** [comment]
> > I think having a default env var set to 1 will be nice and more flexible.
> 
> i'm not sure what you mean. this is the opposite of flexible for me just because internally setting variables that control the build make the script very unflexible to be called from command line

I mean in the script:
- We check if there is a env variable name `WOOCOMMERCE_BLOCKS_PHASE`, if yes, use it.
- If not, we set `WOOCOMMERCE_BLOCKS_PHASE` to 1 as the default value.

By doing so, we still allow modify the block phase by setting env variable, while making it default to product while running the build zip script.

**Comment by samueljseay** [comment]
@dinhtungdu ah yeah I originally considered checking the value of the variable as well, but felt like I'm making an assumption about the broadest usage of the script in development. 

Why do you think in local dev that 1 should be the default for build zip? As I said, setting env variables in a script can lead to surprising behaviour if you didn't read the implementation of the script, so what's the benefit here? a few lines saved in workflows? 

I think its always better to not set any env variables inside scripts that change their behaviour based on the script personally. It increases complexity and forces the script user to know for example what default value we set this env variable to.

**Comment by samueljseay** [comment]
@dinhtungdu I felt unsure about your approach mainly because I still don't like surprising by setting global env flags from inside a script even if it is a fallback. Furthermore I think this variable should go away in future.

That said I think I'm being a bit pedantic as your solution to fallback to a default is a good solution! Thanks for suggesting it. I've added your proposed changes, could you take another look?

**Comment by Aljullu** [comment]
@samueljseay should we mark this PR as fixing https://github.com/woocommerce/woocommerce/issues/43073?

**Comment by samueljseay** [comment]
@Aljullu that issue looks like it discusses some other things such as removing the feature_plugin flag. All this really does is allow the changing of this value from outside the build zip script, so I'm thinking this PR would not close that issue.

**Review by dinhtungdu — APPROVED** [review]
Thanks for addressing my review! This is LGTM!

**Pull request merged**